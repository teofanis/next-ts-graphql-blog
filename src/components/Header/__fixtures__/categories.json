{
  "categories": [
    {
      "name": "React",
      "slug": "react"
    },
    {
      "name": "Redux",
      "slug": "redux"
    },
    {
      "name": "React Native",
      "slug": "react-native"
    }
  ]
}
