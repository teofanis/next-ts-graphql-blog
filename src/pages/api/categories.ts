import { getCategories } from '@services/index'
import type { NextApiRequest, NextApiResponse } from 'next'

export default async function categories(
  req: NextApiRequest,
  res: NextApiResponse
) {
  try {
    const result = getCategories()
    return res.status(200).send(result)
  } catch (error) {
    return res.status(500).send(error)
  }
}
