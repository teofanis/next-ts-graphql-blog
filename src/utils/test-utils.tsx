import { render, RenderOptions } from '@testing-library/react'
import { ReactElement } from 'react'
// Add in any providers here if necessary:

const Providers: React.FC = ({ children }) => {
  return <>{children}</>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'
// override render method
export { customRender as render }
