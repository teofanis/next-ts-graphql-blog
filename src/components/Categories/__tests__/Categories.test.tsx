import { Categories } from '@components/index'
import { render } from '@utils/test-utils'
import React from 'react'
import { categories } from '../__fixtures__/categories.json'

describe('Categories Test', () => {
  it('should render correctly', () => {
    const { container } = render(<Categories categories={categories} />)
    expect(container).toMatchSnapshot()
  })
})
