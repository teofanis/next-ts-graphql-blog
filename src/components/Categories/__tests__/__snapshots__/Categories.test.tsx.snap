// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Test should render correctly 1`] = `
<div>
  <div
    class="bg-white dark:bg-gray-500 shadow-lg rounded-lg p-8 mb-8 pb-12"
  >
    <h3
      class="text-xl mb-8 font-semibold border-b pb-4"
    >
      Categories
    </h3>
    <a
      class="cursor-pointer block pb-3 mb-3 hover:underline"
      data-testid="category-link-container"
      href="/category/react"
    >
      React
    </a>
    <a
      class="cursor-pointer block pb-3 mb-3 hover:underline"
      data-testid="category-link-container"
      href="/category/redux"
    >
      Redux
    </a>
    <a
      class="cursor-pointer block pb-3 mb-3 hover:underline"
      data-testid="category-link-container"
      href="/category/react-native"
    >
      React Native
    </a>
  </div>
</div>
`;
